service/src/NVMLIOGroup.cpp

```diff
@@ -708,9 +701,25 @@ namespace geopm
 
     void NVMLIOGroup::save_control(const std::string &save_path)
     {
+        std::vector<SaveControl::m_setting_s> settings;
+        for (int domain_idx = 0;
+             domain_idx < m_platform_topo.num_domain(GEOPM_DOMAIN_BOARD_ACCELERATOR);
```

Comment:
maybe cache a local copy of the return value to avoid calling num_domain() multiple times